hrms/overrides/company.py

```diff
@@ -132,3 +132,55 @@ def validate_default_accounts(doc, method=None):
 					"The currency of {0} should be same as the company's default currency. Please select another account."
 				).format(frappe.bold(_("Default Payroll Payable Account")))
 			)
+
+
+def handle_linked_docs(doc, method=None):
+	delete_docs_with_company_field(doc)
+	clear_company_field_for_single_doctypes(doc)
+
+
+def delete_docs_with_company_field(doc, method=None):
+	"""
+	Deletes records from linked doctypes where the 'company' field matches the company's name
+	"""
+	company_data_to_be_ignored = frappe.get_hooks("company_data_to_be_ignored") or []
+	for doctype in company_data_to_be_ignored:
+		records_to_delete = frappe.get_all(doctype, filters={"company": doc.name}, pluck="name")
+		if records_to_delete:
+			frappe.db.delete(doctype, {"name": ["in", records_to_delete]})
```

Comment:
Use delete query instead of `frappe.db.delete` because otherwise, it can give a Link Validation error while deleting those records.